### PR TITLE
Background loop feature

### PR DESCRIPTION
**Pull request by mvkv**

### CODE REVIEW

File: flask_server.py [hunk 1]
```diff
@@ -85,12 +84,18 @@ def settings_page():
             if setting_name == "extra_ids":
                 extra_ids = request.form["extra_ids"].splitlines()
                 settings.update("extra_ids", extra_ids)
+            elif setting_name in ["loop_delay", "loop_variation"] and len(request.form[setting_name]) != 0:
+                settings.update(setting_name, int(request.form[setting_name]))
             elif len(request.form[setting_name]) != 0:  # Updates other non-null settings.
                 settings.update(setting_name, request.form[setting_name])
```
Comment (powahftw) [inline flask_server.py]: As we start to have more settings maybe it's better if we update them in a in memory copy and save them to file later! 
Calling settings.update every time it;'s not very efficent.

File: flask_server.py [hunk 2]
```diff
@@ -85,12 +84,18 @@ def settings_page():
...
+        loop_args = {"loop_delay": settings.get("loop_delay"), "loop_variation": settings.get("loop_variation")}
+        scraper_runner.updateDelay(**loop_args)
```
Comment (powahftw) [inline flask_server.py]: Move this out of the loop as you always need it, this way you don't request it two times with settings.get

File: flask_server.py [hunk 3]
```diff
@@ -85,12 +84,18 @@ def settings_page():
...
         logger.info("Updated settings")
     folder_path = settings.get("folder_path")
+    loop_delay = settings.get("loop_delay")
+    loop_variation = settings.get("loop_variation")
     extra_ids = settings.get("extra_ids")
-    return render_template("settings.html", folder_path=folder_path, extra_ids=extra_ids)
+    return render_template("settings.html", settings={"folder_path": folder_path, "loop_delay": loop_delay, "loop_variation": loop_variation}, extra_ids=extra_ids)
```
Comment (powahftw) [inline flask_server.py]: style nit: As settings it's long put extra_ids first in the arguments.

File: settings.py [hunk 4]
```diff
@@ -21,9 +22,11 @@
 
 DEFAULT_VALUES = {
     'scraping_log_file_path': SCRAPING_LOG_FILE_PATH,
-    'folder_path': "ig_media",
+    'folder_path': FOLDER_PATH,
     'extra_ids': [],
     'cached_ids_path': CACHED_IDS_PATH,
+    'loop_delay': 8 * 60 * 60,
```
Comment (powahftw) [inline settings.py]: nit: loop_delay_seconds

File: settings.py [hunk 5]
```diff
@@ -21,9 +22,11 @@
...
+    'loop_variation': 20
```
Comment (powahftw) [inline settings.py]: nit: loop_variation_percentage

File: thread_runner.py [hunk 6]
```diff
@@ -1,21 +1,24 @@
...
         self.output = {}
         self.func = func
         self.th = threading.Thread(target=self.runLoopedFunction, daemon=True)
         self.th.start()
 
     def runLoopedFunction(self):
         while True:
-            time.sleep(self.UPDATE_FREQUENCY)
+            loop_variation = int(self.loop_args["loop_delay"] * (self.loop_args["loop_variation"] / 100))
+            time_delay = self.loop_args["loop_delay"] + random.randint(-loop_variation, loop_variation)
```
Comment (powahftw) [inline thread_runner.py]: create an helper function def waitFor(self): with this logic inside there.

**Review by powahftw — CHANGES_REQUESTED** [review]

File: flask_server.py (hunk 1 above)
Comment (mvkv) [reply flask_server.py]: Done

**Pull request merged**